modules/afsocket/afsocket-dest.c

```diff
@@ -558,6 +558,7 @@ afsocket_dd_setup_writer(AFSocketDestDriver *self)
           log_writer_reopen(self->writer, proto);
         }
     }
+  self->connection_initialized = TRUE;
```

Comment:
This is used to decide whether we want to keep the connection open during reload.

Since the feature is about keeping the connection open (not the writer, or proto), the original place seemed more correct.

Reply:
I might be wrong here, but keeping a broken connection does not seem right.

(The queue won't be lost (acquire_queue() uses `cfg_persist` too.)

Reply:
Yep, I've misunderstood the original code. Even though a broken connection might not be worth keeping, we did that anyway.

Reply:
Sorry for not answering sooner.

Even though you are right that the feature is about keeping the connection across reloads, and now it looks like it's coupled to the LogWriter's initialization only, IMHO this was the case already in the initial commit 006b3d2ac056d29c97a86a177fbb6728a4960661.

You are right that we should rework this kept-alive-across-reload feature to save the writer (thus the connection) only if the connection is established. However, maybe there are other states from the LogWriter that should be saved too even in case of a broken connection, e.g. the `seq_num`.

I'm not planning to do it yet. (Just for transparent communication :) )